tests/loggen/CMakeLists.txt

```diff
@@ -1,11 +1,92 @@
+set(LOGGEN_PLUGIN_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/lib/syslog-ng/loggen")
+
+# ########### loggen_helper static lib ##############
+set(LOGGEN_HELPER_SOURCE
+  loggen_helper.c
+	loggen_helper.h
+	${CMAKE_SOURCE_DIR}/lib/crypto.c
+	${CMAKE_SOURCE_DIR}/lib/compat/openssl_support.c
+  )
+
+add_library(loggen_helper STATIC ${LOGGEN_HELPER_SOURCE})
+
+include_directories(loggen_helper PUBLIC
+  ${CORE_INCLUDE_DIRS}
+  ${CMAKE_CURRENT_SOURCE_DIR}
+  ${GLIB_INCLUDE_DIRS}
+  )
+
+target_compile_definitions(loggen_helper PUBLIC
+  SYSLOG_NG_PATH_LOGGEN_PLUGIN_DIR="${LOGGEN_PLUGIN_INSTALL_DIR}"
+  )
+
+set_target_properties(loggen_helper
+    PROPERTIES VERSION ${SYSLOG_NG_VERSION}
+    SOVERSION ${SYSLOG_NG_VERSION})
+
+install(TARGETS loggen_helper ARCHIVE DESTINATION ${LOGGEN_PLUGIN_INSTALL_DIR})
+
+# ########### loggen_plugin common static lib for plugins #####
+set(LOGGEN_PLUGIN_SOURCE
+  loggen_plugin.c
+	loggen_plugin.h
+  )
+
+add_library(loggen_plugin STATIC ${LOGGEN_HELPER_SOURCE})
+
+include_directories(loggen_plugin PUBLIC
+  ${CORE_INCLUDE_DIRS}
+  ${CMAKE_CURRENT_SOURCE_DIR}
+  ${GLIB_INCLUDE_DIRS}
+  )
+
+target_compile_definitions(loggen_plugin PUBLIC
+  SYSLOG_NG_PATH_LOGGEN_PLUGIN_DIR="${LOGGEN_PLUGIN_INSTALL_DIR}"
+  )
+
+set_target_properties(loggen_plugin
+    PROPERTIES VERSION ${SYSLOG_NG_VERSION}
+    SOVERSION ${SYSLOG_NG_VERSION})
+
+install(TARGETS loggen_plugin ARCHIVE DESTINATION ${LOGGEN_PLUGIN_INSTALL_DIR})
+
+# ########### loggen binary #################
 set(LOGGEN_SOURCE
     loggen.c
-    ${CMAKE_SOURCE_DIR}/lib/crypto.c
-    ${CMAKE_SOURCE_DIR}/lib/compat/openssl_support.c)
+    loggen.h
+    loggen_helper.c
+    loggen_helper.h
```

Comment:
.h files are not needed for add_executable. It would be cleaner to separate those into a different variable and add a separate install command for those.

Reply:
removed